pkg/controller.v1/tensorflow/util.go

```diff
@@ -49,3 +50,18 @@ func ContainChieforMasterSpec(tfJob *tfv1.TFJob) bool {
 	}
 	return false
 }
+
+func CheckTFJobIsNotPending(tfJob *tfv1.TFJob) bool {
+	tfJobStatus := tfJob.Status
+	if tfJobStatus.ReplicaStatuses == nil {
+		tfJobStatus.ReplicaStatuses = map[common.ReplicaType]*common.ReplicaStatus{}
+	}
+
+	createdCount := int32(0)
+	for _, status := range tfJobStatus.ReplicaStatuses {
+		createdCount += status.Active + status.Succeeded + status.Failed
+	}
+	jobCondition := tfJobStatus.Conditions[len(tfJobStatus.Conditions)-1].Type
+
+	return createdCount > 0 && jobCondition != common.JobCreated && jobCondition != common.JobRestarting
```

Comment:
I think if createdCount > 0, it should be not pending. In case all the pods are deleted, we check the job condition further.